### DISCUSSION TITLE
Textrual inversion (embedding) weight

### DISCUSSION THREAD

**Discussion by fractal-fumbler** [comfyanonymous/ComfyUI#360] [Q&A] [answered]
Hello :)

searched for option to set weights (strength) of an embeddings like in a1111: `(embedding:0.7)`, but didn't find.

is it possible in ComfyUI to set this value?

**Comment by comfyanonymous** [comment, marked as answer]
This should work: (embedding:file.pt:1.2)

**Reply by fractal-fumbler** [reply]
thanks, results are different!

and thanks a lot for your work, switched to ComfyUI :)

**Reply by linonetwo** [reply]
@comfyanonymous  Should we include the `.pt` extension?

For example to use https://civitai.com/models/4629/deep-negative-v1x , do we need to use `ng_deepnegative_v1_75t.pt` instead of `ng_deepnegative_v1_75t`? And like `ng_deepnegative_v1_75t.pt, (easynegative.safetensors:1.2)`?

And after put the pt file inside `ComfyUI\models\embeddings`,  we just use it directly in CLIP Text Encode node, no other config?

**Reply by comfyanonymous** [reply]
You can use: embedding:ng_deepnegative_v1_75t or embedding:ng_deepnegative_v1_75t.pt

See: https://comfyanonymous.github.io/ComfyUI_examples/textual_inversion_embeddings/

**Reply by WASasquatch** [reply]
Strangely when I use NG Deepnegative, it says the file can't be found, so I renamed it to NGNegative and then it worked fine. 

> You can use: embedding:ng_deepnegative_v1_75t or embedding:ng_deepnegative_v1_75t.pt
> 
> See: https://comfyanonymous.github.io/ComfyUI_examples/textual_inversion_embeddings/

**Reply by chrisgoringe** [reply]
don't think you can use weights if you don't give the full filename - (embedding:name:1.1) gives error 'name:1.1.pt' not found, but (embedding:name.pt:1.1) works

**Comment by linonetwo** [comment]
Note that when using weight, you must add `()` around them.

## Working example:

all three usages is ok:

```
embedding:ng_deepnegative_v1_75t, (embedding:easynegative.safetensors:1.2), (embedding:bad_prompt_version2:0.8)
```

```
100%|██████████████████████████████████████████████████████████████████████████████████| 28/28 [00:05<00:00,  5.13it/s]
got prompt
```

## Not working example:

These will not work

### Wrong \n line break

```
embedding:ng_deepnegative_v1_75t.pt, embedding:easynegative.safetensors:1.2,embedding:bad_prompt_version2.pt
:0.8
```

```
warning, embedding:easynegative.safetensors:1.2,embedding:bad_prompt_version2.pt does not exist, ignoring
100%|██████████████████████████████████████████████████████████████████████████████████| 28/28 [00:56<00:00,  2.01s/it]
```

### without `()`

```
embedding:easynegative.safetensors:1.2,embedding:bad_prompt_version2.pt
```

```
warning, embedding:easynegative.safetensors:1.2, does not exist, ignoring
warning, embedding:bad_prompt_version2.pt:0.8, does not exist, ignoring
100%|██████████████████████████████████████████████████████████████████████████████████| 28/28 [00:05<00:00,  5.13it/s]
```

**Reply by linonetwo** [reply]
Apart from this, I still don't understand how `()` works. Does `([embedding:ng_deepnegative_v1_75t.pt, embedding:easynegative.safetensors:1.2,embedding:bad_prompt_version2.pt
:0.8](embedding:ng_deepnegative_v1_75t, (embedding:easynegative.safetensors:1.2), (embedding:bad_prompt_version2:0.8)))` the same as example above?

Some people also use `(8k, RAW photo, best quality, masterpiece:1.2)` instead of `8k, RAW photo, best quality, masterpiece:1.2`, and `(shiny skin)` instead of `shiny skin`, what is the difference? Is this the reason? https://www.aidemos.info/what-are-parentheses-and-brackets-in-stable-diffusion/
  will it work in ComfyUI too?
